src/actions/arcActions.ts

```diff
@@ -797,7 +800,20 @@ export function stakeProposal(daoAvatarAddress: string, proposalId: string, pred
       if (amount.lt(minimumStakingFee)) { throw new Error(`Staked less than the minimum: ${Util.fromWei(minimumStakingFee).toNumber()}!`); }
       if (amount.gt(balance)) { throw new Error(`Staked more than than the balance: ${Util.fromWei(balance).toNumber()}!`); }
 
-      const stakeTransaction = await votingMachineInstance.stake({ proposalId, vote: prediction, amount });
+      const key = Arc.TransactionService.generateInvocationKey('Stake');
```

Comment:
Would it make sense to put `const key = Arc.TransactionService.generateInvocationKey('Stake'); ` inside of `onPendingTransactions`?

Reply:
done in: https://github.com/daostack/alchemy/pull/61/commits/366f5bd5e619d50fa3ea85893029b5ec18f705ad